openapi.yaml

```diff
@@ -0,0 +1,787 @@
+    parameters:
+      - name: "permissionSetId"
+        in: "path"
+        description: "Id of PermissionSet to return"
+        required: true
+        schema:
+          type: "string"
+    get:
+      operationId: readPermissionSet
+      description: Access an existing Permission Set
+      tags:
+        - "permissionsets"
+      responses:
+        "200":
+          description: "Display Permission Set"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/PermissionSetResponse"
+        "404":
+          $ref: "#/components/responses/NotFound"
+        "500":
+          $ref: "#/components/responses/ServerError"
+    put:
+      operationId: updatePermissionSet
+      description: Update an existing Permission Set
+      tags:
+        - "permissionsets"
+      requestBody:
+        required: true
+        content:
+          'application/json':
+            schema:
+              $ref: "#/components/schemas/PermissionSetRequest"
+      responses:
+        "200":
+          description: "Display updated Permission Set"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/PermissionSetResponse"
+        "400":
+          $ref: "#/components/responses/BadInput"
+        "404":
+          $ref: "#/components/responses/NotFound"
+        "500":
+          $ref: "#/components/responses/ServerError"
+    delete:
+      operationId: deletePermissionSet
+      description: Delete an existing Permission Set
+      tags:
+        - "permissionsets"
+      responses:
+        "200":
+          $ref: "#/components/responses/Deleted"
+        "404":
+          $ref: "#/components/responses/NotFound"
+        "500":
+          $ref: "#/components/responses/ServerError"
+  /permissions:
+    get:
+      operationId: getPermissions
+      tags:
+        - "permissionsets"
+      responses:
+        "200":
+          description: "Display available permissions"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/PermissionsResponse"
+  /grants:
+    get:
+      operationId: listGrants
+      tags:
+        - "grants"
+      parameters:
+        - $ref: "#/components/parameters/pagingOffset"
+        - $ref: "#/components/parameters/pagingLimit"
+        - name: "permissionSetId"
+          in: "query"
+          description: Limit to the specified permissionset
+          required: false
+          schema:
+            type: string
+        - name: "securableId"
+          in: "query"
+          description: Limit to the specified securable
+          required: false
+          schema:
+            type: string
+      responses:
+        "200":
+          description: "Display retrieved Grants"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/GrantsResponse"
+    post:
```

Comment:
Regular users can view grants that they own, meaning they must have read access to the DAM.  However they should not have write access by default.  Therefore 403 should be a valid response for creation here